### PR TITLE
Alias analysis support for macros

### PR DESCRIPTION
**Pull request by mwu-tow**
### Pull Request Description
This PR extends alias analysis support for macro-using code, including special support for lambda expressions.
Also, it fixes already standing omission where definition body identifier usage was not properly covered in the results. (We had assumed that we won't need it for connections — and we haven't for connections per se, but we need for utilities for connection ops).

Also, as I encountered a number of issues with parser, I've made sure that they are reported and created a test cases for them to easily keep track as we fix them.

Marking this as draft, as code is known to fail tests, because the location of the definition name is wrongly reported. This will be fixed soon. (this is the FIXME note in the deifnition processor)

### Checklist
Please include the following checklist in your PR:

- [x] The documentation has been updated if necessary.
- [x] All code conforms to the [Rust](https://github.com/luna/enso/blob/master/doc/rust-style-guide.md), [Scala](https://github.com/luna/enso/blob/master/doc/scala-style-guide.md), [Java](https://github.com/luna/enso/blob/master/doc/java-style-guide.md) or [Haskell](https://github.com/luna/enso/blob/master/doc/haskell-style-guide.md) style guides as appropriate.
- [x] All code has been tested where possible.

### CODE REVIEW

File: src/rust/ide/ast/impl/src/crumbs.rs [hunk 1]
```diff
@@ -368,7 +368,10 @@ macro_rules! impl_crumbs {
             fn get(&self, crumb:&Self::Crumb) -> FallibleResult<&Ast> {
                 match (self,crumb) {
                     $((Shape::$id(shape),Crumb::$id(crumb)) => shape.get(crumb),)*
-                    _ => Err(MismatchedCrumbType.into())
+                    _ => {
+                        println!("Mismatch between crumb={:?} when accessing AST {}", crumb, self.repr());
```
Comment (mwu-tow) [inline src/rust/ide/ast/impl/src/crumbs.rs:372]: Debug print to be removed

File: src/rust/ide/src/controller/graph.rs [hunk 2]
```diff
@@ -373,7 +373,7 @@ impl Handle {
     /// resolution in the code in this graph.
     pub fn used_names(&self) -> FallibleResult<Vec<LocatedName>> {
         let def    = self.graph_definition_info()?;
-        if let Ok(block)  = ast::known::Block::try_from(def.body()) {
+        if let Ok(block)  = ast::known::Block::try_from(*def.body()) {
```
Comment (farmaazon) [inline src/rust/ide/src/controller/graph.rs:376]:
```suggestion
        if let Ok(block) = ast::known::Block::try_from(*def.body()) {
```

File: src/rust/ide/src/double_representation/alias_analysis.rs [hunk 3]
```diff
@@ -235,6 +235,19 @@ impl AliasAnalyzer {
         self.in_location_of(&located_ast, |this| this.process_ast(located_ast.item.borrow()))
     }
 
+    /// Processes subtrees of the given AST denoted by given crumb sequence items.
```
Comment (farmaazon) [inline src/rust/ide/src/double_representation/alias_analysis.rs:238]:
```suggestion
    /// Processes subtrees of the given AST denoted by given crumbs
```

File: src/rust/ide/src/double_representation/alias_analysis.rs [hunk 4]
```diff
@@ -407,6 +452,32 @@ mod tests {
...
                 foo x = 2
                 foo »+« 1",
+
+            // === Macros Match ===
+            "a -> a",
+            "a -> »b«",
+            "»A« -> »b«",
+            "a -> A -> a",
+            "a -> a -> A",
+            "x»,«y -> »B«",
```
Comment (farmaazon) [inline src/rust/ide/src/double_representation/alias_analysis.rs:462]:
```suggestion
            "x»,«y -> »B«",
            "x»,«y -> y",
```

**Pull request merged**